symplyphysics/laws/chemistry/potential_energy_models/hard_spheres_potential.py

```diff
@@ -0,0 +1,48 @@
+from sympy import Eq, Piecewise, S
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    print_expression,
+    validate_input,
+    validate_output,
+)
+
+# Description
+## Hard spheres are widely used as model particles in the statistical mechanical theory, defined as
+## impenetrable spheres that cannot overlap in space, which mimics extremely strong repulsion that
+## atoms and molecules experience at very close distances
+
+# Law: U(r) = infinity if r <= sigma else 0
+## U - hard spheres potential
+## r - distance between particles
+## sigma - diameter of sphere
+
+potential = Symbol("potential", units.energy)
+distance = Symbol("distance", units.length)
+sphere_diameter = Symbol("sphere_diameter", units.length)
+
+law = Eq(
+    potential,
+    Piecewise((S.Infinity, distance <= sphere_diameter), (0, distance > sphere_diameter)),
```

Comment:
Here we should get infinite energy, not some dimensionless infinity.
Try `S.Infinity * units.energy` or `Quantity(S.Infinity, units.joule)`

Reply:
Oh, I see now

Reply:
![image](https://github.com/blackyblack/symplyphysics/assets/115647936/10b24edf-b17c-49dc-96ed-cba253cbabb6)
Still doesn't work with infinity

Reply:
`S.Infinity * units.joule` doesn't work either

Reply:
I've fixed it.

Reply:
Now I see how to manage it, thanks

Reply:
Though this caused a problem in the compressibility factor law since `sympy` now doesn't know that there's infinity in that quantity of ours in this law

Reply:
The problem goes away if I substitute that quantity with proper `sympy.S.Infinity`

Reply:
Yes, I see that CI is failing. I will look into it.

Reply:
Done. Take a look.